File: 2018/pset1/credit/credit.c
#include <cs50.h>
#include <stdio.h>
#include <math.h>

int main(void)
{
    long long n;
    do
    {
        n = get_long_long("Number: ");
    }
    while (n < 0);

    int type = 0;
    if (n >= pow(10, 14) && n < pow(10, 15))
    {
        int am = n / pow(10, 13);
        if (am == 34 || am == 37)
        {
            type = 1;
        }
    }
    else if (n >= pow(10, 12) && n < pow(10, 13))
    {
        int i = n / pow(10, 12);
        if (i == 4)
        {
            type = 2;
        }
    }
    else if (n >= pow(10, 15) && n < pow(10, 16))
    {
        int j = n / pow(10, 14);
        if (j >= 51 && j <= 55)
        {
            type = 3;
        }
        else if (j >= 40 && j <= 49)
        {
            type = 2;
        }
    }

    if (type > 0)
    {
        int sum = 0;
        while (n > 0)
        {
            sum += (n % 10);
            n /= 10;

            int bytwo = (n % 10) * 2;
            if (bytwo >= 10)
            {
                bytwo = (bytwo % 10) + 1;
            }
            sum += bytwo;
            n /= 10;
        }
        if (sum % 10 != 0)
        {
            type = 0;
        }
    }

    switch (type)
    {
        case 0:
            printf("INVALID\n");
            break;
        case 1:
            printf("AMEX\n");
            break;
        case 2:
            printf("VISA\n");
            break;
        case 3:
            printf("MASTERCARD\n");
            break;
    }

}

File: 2018/pset6/credit/credit.py
from cs50 import get_int

while True:
    n = get_int("Number: ")
    if n >= 0:
        break

typ = 0
if len(str(n)) == 15:
    am = str(n)[0:2]
    if am == "34" or am == "37":
        typ = 1
elif len(str(n)) == 13:
    if str(n)[0:1] == "4":
        typ = 2
elif len(str(n)) == 16:
    j = n // 10 ** 14
    if j >= 51 and j <= 55:
        typ = 3
    elif j >= 40 and j <= 49:
        typ = 2

if typ > 0:
    s = 0
    while n > 0:
        s += n % 10
        n //= 10

        bytwo = (n % 10) * 2
        if bytwo >= 10:
            bytwo = bytwo % 10 + 1
        s += bytwo
        n //= 10

    if s % 10 != 0:
        typ = 0

if typ == 0:
    print("INVALID")
elif typ == 1:
    print("AMEX")
elif typ == 2:
    print("VISA")
elif typ == 3:
    print("MASTERCARD")